strax/plugins/plugin.py

```diff
@@ -667,13 +668,27 @@ def do_compute(self, chunk_i=None, **kwargs):
         )
         subruns = self._check_subruns_uniqueness(kwargs, {k: v.subruns for k, v in kwargs.items()})
 
-        kwargs = {k: v.data for k, v in kwargs.items()}
+        _kwargs = {k: v.data for k, v in kwargs.items()}
         if self.compute_takes_chunk_i:
-            kwargs["chunk_i"] = chunk_i
+            _kwargs["chunk_i"] = chunk_i
         if self.compute_takes_start_end:
-            kwargs["start"] = start
-            kwargs["end"] = end
-        result = self.compute(**kwargs)
+            _kwargs["start"] = start
+            _kwargs["end"] = end
+        result = self.compute(**_kwargs)
+        del _kwargs
```

Comment:
Delete `_kwargs` because it will keep a reference between `_kwargs` and `kwargs[k].data`.